DESIGN.md

```diff
@@ -235,7 +244,6 @@ CK | ts | timestamp rounded down by granularity
     - It's important that the aggregate data row be `UPDATE`d rather than `INSERT`ed to accommodate late arrival handling, describe below. To ruin the surprise, the aggregated metric values are upserted in order to allow for re-calculation later and re-upserting the new value.
     - The TTL used for each update-batch will be determined by the retention duration configured for that granularity. For example, 5m granularity might configured with 14 days retention and 1h with 1 year retention.
 - The `metric_names` table is also upserted with the original raw metric's name and the aggregators identified during the aggregation process. For example, a gauge with the name `cpu_idle` could start with an `aggregators` of only `{'raw'}`, but after aggregation processing would become `{'raw','min','max','avg'}`
-- Finally, the pending downsample set row is deleted to track that it has been processed
 
 #### Processing late arrivals
 
```

Comment:
lines 252-260 below are out of date with respect to redis.  They should look something like this:

```
- No metrics at all showed up for a given tenant+series-set during a time slot. This case is handled by the design above since:
    - The redis ingesting/pending entries would not have been created at the original time
    - When those metrics arrive later, the redis entries are created
    - Once the readiness of the downsample set is satisfied it is processed as normal
- A subset of metrics showed up for a given tenant+series-set at the original time
    - The downsample processor won't know that there's a subset of the expected metrics; however, that's fine because all of the aggregations are mathematically correct for what values are present. For example, the "count" aggregation would be mathematically accurate, even if technically lower than a user would expect.
    - When the remaining metrics arrive later, the redis entries are **re-created**
    - Once the readiness of the downsample set is satisfied the downsample processor will pick up on the redis entries at the next scheduled time, the re-query of that time slot's raw data will now retrieve all expected metrics entries. _This assumes the raw data for that queried time slot has not been TTL'ed away._
    - The downsample processor will aggregate the granularities entries as described above and again upsert/UPDATE the resulting update-batches as usual. With the UPDATE the "partial" aggregation values will be replaced by the "complete" aggregation values
```

Reply:
Thanks! I had missed that ending spot when redis-ifying the design document.